base/markdown/GitHub/table.jl

```diff
@@ -0,0 +1,168 @@
+type Table
+    rows::Vector{Vector{Any}}
+    align
+end
+
+function github_table(stream::IO, md::MD, config::Config)
+    withstream(stream) do
+        rows = Any[]
+        n = 0
+        align = :r # default is to align right
+        while !eof(stream)
+            n += 1
+            pos = position(stream)
+            skipwhitespace(stream)
+            line = readline(stream) |> chomp
+
+            if n == 1
+                pipe_border = line[1] == '|'
+                if !('|' in line)
+                    return false
+                end
+            end
+
+            row = map(strip, split(line, "|"))
+            if pipe_border
+                if row[1] == row[end] == ""
+                    row = row[2:end-1]
+                else
+                    return false
+                end
+            end
+
+            if n == 2 && all(['-' in r && issubset(Set(r), Set(" -:"))
+                             for r in row])
+                # handle possible --- line
+                align = Symbol[]
+                for r in row
+                    if r[1] == ':'
+                        if r[end] == ':'
+                            push!(align, :c)
+                        else
+                            push!(align, :l)
+                        end
+                    else
+                        if r[end] == ':'
+                            push!(align, :r)
+                        else
+                            # default is align right
+                            push!(align, :r)
+                        end
+                    end
+                end
+
+            elseif n == 1 || length(rows[1]) == length(row)
+                push!(rows, map(x -> parseinline(x, config), row))
+            elseif length(row) > 1
+                seek(stream, pos)
```

Comment:
There's a strong practice in Markdown.jl of not handling the stream position explicitly. I guess it's not essential, but it would still be nice to see row parsing factored out to address that